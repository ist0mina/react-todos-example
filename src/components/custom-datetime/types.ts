export interface CustomDatetimeProps {
    controlId: string;
    title?: string;
    selected?: Date;
    disabled?: boolean;    
    onChange?: (date: Date | null) => void;    
};
