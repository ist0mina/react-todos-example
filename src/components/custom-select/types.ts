export type OptionSelect = {
    key: string;
    value: string;
};

export interface CustomSelectProps {
    controlId: string;
    title?: string;
    data?: OptionSelect[];
    selected?: string;
    disabled?: boolean;
    onChange?: (event: React.ChangeEvent<HTMLSelectElement>) => void;
};