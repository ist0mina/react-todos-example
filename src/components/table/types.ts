export interface TableBasicProps {
    hasNNCol?: boolean;
    hasControlCol?: boolean;
}

export type TableColumItem = {
    title: string;
}

export interface TableHeaderProps extends TableBasicProps {
    columns: TableColumItem[];
}

export type TableCellItem = {
    text: string;
}

export type TableRowItem = {
    cells: TableCellItem[];
}

export interface TableBodyProps extends TableBasicProps {
    rows: TableRowItem[]; 
}

export interface TableRowProps extends TableBasicProps {
    cols: TableCellItem[];
    index: number;
}

export type SimpleTableProps = TableHeaderProps & TableBodyProps;