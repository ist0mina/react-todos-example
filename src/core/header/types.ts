export interface HeaderStateProps {    
    pathname: string;
};

export type HeaderProps = HeaderStateProps;