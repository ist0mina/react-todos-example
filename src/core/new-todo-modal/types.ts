import { ToggleFieldType } from '../../features/toggle/types';
import { TodoItem } from '../../features/todos/types';

export interface NewTodoModalStateProps {
    show?: boolean;
    maxId?: number;
}

export interface NewTodoModalDispatchProps {
    toggle?: (field: ToggleFieldType) => void;
    addTodo?: (todo: TodoItem) => void;
}

export interface NewTodoModalOwnProps {
    handleClose: () => void;
    handleOk: (text: string) => void;
}

export type NewTodoModalProps = NewTodoModalStateProps & NewTodoModalDispatchProps & NewTodoModalOwnProps;