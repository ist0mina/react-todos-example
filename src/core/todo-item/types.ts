import { TodoStatusType } from '../../features/todos/types';

export interface TodoItemStateProps {    
    text: string;
    open: boolean;
    status: TodoStatusType;
    dateCreated: Date;
    dateLastChanged?: Date;
};

export interface TodoItemDispatchProps {
    toggle: (todo: number) => void;
    openTodo: (id: number) => void;
}

export interface TodoItemOwnProps {
    id: number;
};

export type TodoItemProps = TodoItemStateProps & TodoItemDispatchProps & TodoItemOwnProps;