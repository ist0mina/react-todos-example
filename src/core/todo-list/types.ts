export interface TodoListStateProps {
    ids: number[];
};

export type TodoListProps = TodoListStateProps;