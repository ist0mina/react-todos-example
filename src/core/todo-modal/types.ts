import { ToggleFieldType } from '../../features/toggle/types';
import { TodoItem } from '../../features/todos/types';

export interface TodoModalStateProps extends TodoItem {
    show?: boolean;            
}

export interface TodoModalDispatchProps {
    toggle: (field: ToggleFieldType) => void;
    saveTodo: (todo: TodoItem) => void;
};

export type TodoModalProps = TodoModalStateProps & TodoModalDispatchProps;