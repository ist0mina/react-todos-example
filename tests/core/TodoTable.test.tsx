import { TodoTableContainer } from '../../src/core/todo-table';

import { TodoActionTypes } from '../../src/features/todos/types';

import { mockTodos } from '../../src/services/mockData';

import { reduxify, makeMountrender, snapshotify, mockStore } from '../utils';

describe('<TodoTableContainer>', () => {
    it('matches snapshots', () => {        
        const component = reduxify({ Component: TodoTableContainer });
        const wrapper = makeMountrender(component)();

        expect(snapshotify(wrapper)).toMatchSnapshot();
    });
    
    it('table has rows equal to data length', () => {
        const store = mockStore();
        store.dispatch({ type: TodoActionTypes.LOAD_TODOS_SUCCESS, payload: mockTodos});
        
        const component = reduxify({ Component: TodoTableContainer, store });
        const wrapper = makeMountrender(component)();
        
        expect(wrapper.find('tbody').find('tr').length).toBe(mockTodos.length);
    });

    
});