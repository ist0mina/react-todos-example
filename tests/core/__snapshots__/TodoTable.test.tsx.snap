// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoTableContainer> matches snapshots 1`] = `"<table class=\\"table table-striped table-bordered table-hover\\"><thead><tr><th class=\\"td-control\\"></th><th class=\\"td-nn\\">#</th><th>Text</th><th>Status</th></tr></thead><tbody></tbody></table>"`;
